pkg/alerts/nrql_conditions.go

```diff
@@ -350,17 +351,18 @@ type NrqlAlertCondition struct {
 
 // NrqlCondition represents a New Relic NRQL Alert condition.
 type NrqlCondition struct {
-	Enabled             bool              `json:"enabled"`
-	IgnoreOverlap       bool              `json:"ignore_overlap,omitempty"`
-	ExpectedGroups      int               `json:"expected_groups,omitempty"`
-	ID                  int               `json:"id,omitempty"`
-	ViolationCloseTimer int               `json:"violation_time_limit_seconds,omitempty"`
-	Name                string            `json:"name,omitempty"`
-	Nrql                NrqlQuery         `json:"nrql,omitempty"`
-	RunbookURL          string            `json:"runbook_url,omitempty"`
-	Terms               []ConditionTerm   `json:"terms,omitempty"`
-	Type                string            `json:"type,omitempty"`
-	ValueFunction       ValueFunctionType `json:"value_function,omitempty"`
+	Enabled             bool               `json:"enabled"`
+	IgnoreOverlap       bool               `json:"ignore_overlap,omitempty"`
+	ExpectedGroups      int                `json:"expected_groups,omitempty"`
+	ID                  int                `json:"id,omitempty"`
+	ViolationCloseTimer int                `json:"violation_time_limit_seconds,omitempty"`
+	Name                string             `json:"name,omitempty"`
+	Nrql                NrqlQuery          `json:"nrql,omitempty"`
+	RunbookURL          string             `json:"runbook_url,omitempty"`
+	Terms               []ConditionTerm    `json:"terms,omitempty"`
+	Type                string             `json:"type,omitempty"`
+	ValueFunction       ValueFunctionType  `json:"value_function,omitempty"`
+	EntityGUID          *common.EntityGUID `json:"entity_guid,omitempty"`
```

Comment:
:memo: This is a pointer to `common.EntityGUID` because the `NrqlCondition` struct is used to create conditions too.

When creating conditions, a user will not have the EntityGUID available, and it should be omitted completely from the JSON object.